Library/Homebrew/cleaner.rb

```diff
@@ -31,7 +31,11 @@ def clean
     # Get rid of any info 'dir' files, so they don't conflict at the link stage
     Dir.glob(@f.info/"**/dir").each do |f|
       info_dir_file = Pathname(f)
-      observe_file_removal info_dir_file if info_dir_file.file? && !@f.skip_clean?(info_dir_file)
+      next if info_dir_file == Pathname("#{@f.info}/#{@f.name}/dir")
```

Comment:
Can this get a new comment? It's not clear why this is fine to not remove.

Reply:
```suggestion
      next if info_dir_file == @f.info/@f.name/"dir"
```
I think we can also simplify it to this.

Reply:
Yes but: I still don't understand the "why" from this code 😅

Reply:
Oh, I agree with adding a comment explaining why. I'm just allergic to unnecessary interpolation, since they're harder to read and makes code less compact.